crates/matrix-sdk-base/src/session.rs

```diff
@@ -20,6 +20,22 @@ use serde::{Deserialize, Serialize};
 
 /// A user session, containing an access token and information about the
 /// associated user account.
+///
+/// # Example
+///
+/// ```
+/// use matrix_sdk_base::Session;
+/// use ruma::UserId;
+/// use std::str::FromStr;
+///
+/// let session = Session {
+///     access_token: String::from("abcdefgh012345678"),
```

Comment:
Let's copy the example from `Client::restore_login()` instead, that one doesn't have the unwraps and I think `from_str()` is quite uncommon.

If you think we should showcase how to dynamically create the user ID instead, this one would be preferable:

```
/// use matrix_sdk_base::Session;
/// use ruma::UserId;
///
/// # fn main() -> anyhow::Result<()> {
/// let session = Session {
///     access_token: String::from("My-Token"),
///     user_id: UserId::try_from("@example:localhost")?,
///     device_id: "DEVICEID".into()
/// };
/// # Ok(())
/// # }
```

Don't forget to add `anyhow = "1.0.45"` to the `Cargo.toml` file of the base crate.

Reply:
Thanks, I added a commit based on the `Client::restore_login()` example.